src/utils/yarn.js

```diff
@@ -1,10 +1,45 @@
 // @flow
 import includes from 'array-includes';
 
+import type { Dependency, configDependencyType } from '../types';
 import type Package from '../Package';
 import * as processes from './processes';
 import * as fs from '../utils/fs';
 
+export async function add(
+  pkg: Package,
+  dependencies: Array<Dependency>,
+  type?: configDependencyType
+) {
+  const spawnArgs = ['add'];
+  if (!dependencies.length) return;
+
+  dependencies.forEach(dep => {
+    if (dep.version) {
+      spawnArgs.push(`${dep.name}@${dep.version}`);
+    } else {
+      spawnArgs.push(dep.name);
+    }
+  });
+
+  if (type) {
+    const typeToFlagMap = {
+      dependencies: '',
+      devDependencies: '--dev',
+      peerDependencies: '--peer',
+      optionalDependencies: '--optional'
```

Comment:
I think it'd be better to have this conversion happened up in the commands rather than in the utilities

Reply:
I thought we said in the last PR that the utility should convert js options to cli options and commands should be converting from flags to js options?
https://github.com/pyarn/pyarn/pull/17/files#r138537987

Reply:
Ohhh, I read this code backwards, I thought you were converting flags to types not types to flags

Reply:
Yea, I might refactor using Keys later https://flow.org/en/docs/types/utilities/#toc-keys